commit b13c8a01a62df2e1a50aa19edca84cf31cef96f0
Author: sveinung <sveinung@a0f10bec-cc02-0410-94fc-a9cfff90b4cd>
Date:   Tue Jun 28 10:39:31 2016 +0000

    Allow unlikely to win attacks.
    
    Round the "Attack" action probability up. If an action probability is
    rounded down to 0% the action is seen as disabled.
    
    Reported by Marko Lindqvist <cazfi>
    
    See bug #24802
    
    git-svn-id: svn://svn.gna.org/svn/freeciv/trunk@33081 a0f10bec-cc02-0410-94fc-a9cfff90b4cd

diff --git a/common/actions.c b/common/actions.c
index e599b78..9c3703f 100644
--- a/common/actions.c
+++ b/common/actions.c
@@ -15,6 +15,8 @@
 #include <fc_config.h>
 #endif
 
+#include <math.h> /* ceil */
+
 /* utility */
 #include "astring.h"
 
@@ -2423,7 +2425,9 @@ action_prob(const enum gen_action wanted_action,
       if (can_player_see_unit(actor_player, defender_unit)) {
         double unconverted = unit_win_chance(actor_unit, defender_unit);
 
-        chance = (int)((double)200 * unconverted);
+        /* Action is seen as disabled by anyone that relies on action
+         * probability if it is rounded down to 0%. */
+        chance = ceil((double)200 * unconverted);
       } else if (known == TRI_YES) {
         known = TRI_MAYBE;
       }
